test/e2e/upgradetest/upgrade_test.go

```diff
@@ -312,6 +312,94 @@ func testBackupForOldClusterWithBackupPolicy(t *testing.T, bp *spec.BackupPolicy
 	}
 }
 
+// TestDisasterRecovery tests if the new operator could do disaster recovery from backup of the old cluster.
+func TestDisasterRecovery(t *testing.T) {
+	t.Run("Recover from PV backup", func(t *testing.T) {
+		testDisasterRecoveryWithBackupPolicy(t, e2eutil.NewPVBackupPolicy())
+	})
+	t.Run("Recover from S3 backup", func(t *testing.T) {
+		testDisasterRecoveryWithBackupPolicy(t, e2eutil.NewS3BackupPolicy())
+	})
+}
+
+func testDisasterRecoveryWithBackupPolicy(t *testing.T, bp *spec.BackupPolicy) {
+	err := testF.CreateOperator()
+	if err != nil {
+		t.Fatalf("failed to create operator: %v", err)
+	}
+	defer func() {
+		err := testF.DeleteOperator()
+		if err != nil {
+			t.Fatalf("failed to delete operator: %v", err)
+		}
+	}()
+	err = k8sutil.WaitEtcdTPRReady(testF.KubeCli.CoreV1().RESTClient(), 3*time.Second, 30*time.Second, testF.KubeNS)
+	if err != nil {
+		t.Fatalf("failed to see cluster TPR get created in time: %v", err)
+	}
+
+	bp.BackupIntervalInSecond = 60 * 60 * 24 // long enough that no backup was made automatically
```

Comment:
Edit: It's actually 3600s when we create the backup policy which is still fine for our purpose.